homeassistant/components/solar_frontier/sensor.py

```diff
@@ -0,0 +1,200 @@
+"""Solart Frontier Inverter interface."""
+from __future__ import annotations
+
+from datetime import date, datetime
+import logging
+
+import pysolarfrontier
+import voluptuous as vol
+
+from homeassistant.components.sensor import (
+    PLATFORM_SCHEMA,
+    STATE_CLASS_MEASUREMENT,
+    SensorEntity,
+)
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    DEVICE_CLASS_ENERGY,
+    ENERGY_KILO_WATT_HOUR,
+    EVENT_HOMEASSISTANT_START,
+    EVENT_HOMEASSISTANT_STOP,
+)
+from homeassistant.core import CALLBACK_TYPE, callback
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import async_call_later
+from homeassistant.util import dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_INTERVAL = 60
+MAX_INTERVAL = 500
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {vol.Required(CONF_HOST): cv.string, vol.Optional(CONF_NAME): cv.string}
+)
```

Comment:
We do not accept new/changed platform configuration (https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md)

This needs to implement a config flow (https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md)